server/controllers/events.js

```diff
@@ -35,4 +35,22 @@ module.exports = {
       console.log(err);
     }
   },
+  deleteEvent: async (req, res) => {
+    try {
+      const eventId = req.params.id;
+      await Event.deleteOne({ _id: eventId });
+      res.json({ message: 'Event deleted' });
+    } catch (error) {
+      console.error(error);
+    }
+  },
```

Comment:
a few notes here (also applies to `deleteAllEvents` below):
* These routes aren't actually being gated by authed users - i.e. anyone, regardless of whether they're logged in or not, can hit these routes and delete events.  I see there's [middleware](https://github.com/Caleb-Cohen/Together/blob/55461bf0579b93aa3f8e73244e91287c5bcf55e6/server/middleware/auth.js#L2) to handle this already in the repo but it doesn't seem like it's being used anywhere.
* Similarly, even though we validate the user on the client, we need to validate on the server as well.  Clients should never be trusted to vet actions since any request can be made from the client.
* A controller should always either respond to the request, or throw an error if there is error handling middleware somewhere further down the chain.  In this case, if it errors, the request will never get a response, then time out after many seconds, which is generally undesirable UX - if it fails, let them know immediately by sending a non-200 HTTP status code.

Reply:
That middleware you listed is a left-over remnant of BuB. If I recall it's not useable in its current form because of the structure of react? I may be wrong, but some sort of fix should be implemented. 

Is it enough to just ensure the user is authenticated? Should we not also run the `authorCheck` on the backend?

Reply:
@timmyichen I added ensure auth to the creation and deletion of events.

I also added an event check to ensure that there that the event requested for deletion matches the request user and the event object user. 

Any other feedback before I merge?